File: vdp-ruby/test/vctc/consumer_rules_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'

class ConsumerRulesTest < Test::Unit::TestCase
  def setup
    @config = YAML.load_file('configuration.yml')
    @cardRegisterData ='''{
      "primaryAccountNumber": '''+ @config['vctcTestPan'] + '''
    }'''
  end
  
  def test_registerACard
    base_uri = 'vctc/'
    resource_path = 'customerrules/v1/consumertransactioncontrols'
    url = "#{@config['visaUrl']}#{base_uri}#{resource_path}"
    user_id = @config['userId']
    password = @config['password']
    key_path = @config['key']
    cert_path = @config['cert']
    # Passing correlation id header (x-correlation-id) is optional while making API calls.  
    correlation_id = (0...12).map { (48 + rand(10)).chr }.join
    headers = {'content-type'=> 'application/json', 'accept'=> 'application/json', 'x-correlation-id'=> "#{correlation_id}_SC"}
    puts " Register a Card test"
    puts url
    puts @cardRegisterData
    begin
      response = RestClient::Request.execute(
      :method => :post,
      :url => url,
      :headers => headers,
      :payload => @cardRegisterData,
      :user => user_id,
      :password => password,
      :ssl_client_key => OpenSSL::PKey::RSA.new(File.read(key_path)),
      :ssl_client_cert =>  OpenSSL::X509::Certificate.new(File.read(cert_path))
      )
      puts "Response Status : #{response.code.to_s}"
      puts "Response Headers : " 
      for header,value in response.headers
      puts "#{header.to_s} : #{value.to_s}"
      end
      puts "Response Body : " + JSON.pretty_generate(JSON.parse(response.body))
    rescue RestClient::ExceptionWithResponse => e
      puts "Response Status : #{e.response.code.to_s}"
      puts "Response Headers : " 
      for header,value in e.response.headers
      puts "#{header.to_s} : #{value.to_s}"
      end
      puts "Response Body : " + JSON.pretty_generate(JSON.parse(e.response.body))
      assert(false, "Register a Card test failed")
    end
  end
end

File: vdp-node/test/vctc/consumerrulestest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var abstractVisaAPIClient = require('../../libs/abstractvisapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Visa Consumer Transaction Controls Consumer Rules Test', function() {
	var cardRegisterData = JSON.stringify({
		  "primaryAccountNumber": config.vctcTestPan
	});
	
	it('Register A Card Test',function(done) {
		this.timeout(10000);
		var baseUri = 'vctc/';
		var resourcePath = 'customerrules/v1/consumertransactioncontrols';
		abstractVisaAPIClient.doMutualAuthRequest(baseUri + resourcePath, cardRegisterData, 'POST', {}, 
		function(err, responseCode) {
		    if(!err) {
		    	assert.equal(responseCode, 200);
		    } else {
		        assert(false);
		    }
		    done();
		});
	});
});